web-api/src/personalization/knn.rs

```diff
@@ -118,13 +123,12 @@ async fn merge_knn_searchs(
             Ok(documents) => {
                 // the same document can be returned with different elastic scores, hence the
                 // documents are deduplicated and only the highest score is retained for each
-                for document in documents {
+                for (idx, mut document) in documents.into_iter().enumerate() {
+                    document.score = rrf_score(DEFAULT_RRF_K, idx, 1.0);
```

Comment:
@acrrd  I realized this can also use rrf. But I'm not sure if I should switch it back to select the max score instead.

In general I realized that if we had time could could make sense to move the multiple cois search + merge part into the elastic code and apply it before fetching additional information from postgres.